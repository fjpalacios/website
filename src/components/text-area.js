import * as React from 'react'
import './text-area.scss'

const TextArea = ({ text }) => {
  return (
    <div className="text-area">
      <div className="text-area__content" dangerouslySetInnerHTML={{ __html: text }}></div>
    </div>
  )
}

export default TextArea
