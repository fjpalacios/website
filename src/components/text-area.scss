.text-area {
  margin-bottom: 20px;
  padding: 0 20px;

  @include print {
    margin-bottom: 10px;
    padding: 0 10px;
  }

  &__content {
    p {
      text-align: justify;
      text-justify: inter-word;
    }
  }
}
