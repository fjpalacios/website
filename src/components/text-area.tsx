import './text-area.scss'
import React, { FunctionComponent, ReactElement } from 'react'

type textAreaProps = {
  text: string
}

export const TextArea: FunctionComponent<textAreaProps> = ({
  text,
}): ReactElement => {
  return (
    <div className="text-area">
      <div
        className="text-area__content"
        dangerouslySetInnerHTML={{ __html: text }}
      ></div>
    </div>
  )
}
