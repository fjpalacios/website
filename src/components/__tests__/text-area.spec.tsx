import createComponentWithIntl from '../../../__helpers__/i18n-component'
import React from 'react'
import { TextArea } from '../text-area'

describe('TextArea', () => {
  it('should render properly', () => {
    const textArea = createComponentWithIntl(<TextArea text="Foo" />)
    expect(textArea).toMatchSnapshot()
  })

  it('should show the passed prop as HTML content', () => {
    const { container } = createComponentWithIntl(<TextArea text="Foo" />)
    const title =
      container.getElementsByClassName('text-area__content')[0].textContent
    expect(title).toBe('Foo')
  })
})
