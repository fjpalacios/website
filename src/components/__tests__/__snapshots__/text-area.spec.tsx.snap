// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render properly 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={
    Object {
      "menu": Object {
        "switchTheme": "Switch Theme",
        "toEnglish": "Switch to English language",
        "toSpanish": "Switch to Spanish language",
      },
      "metaData": Object {
        "description": "Francisco Javier Palacios Pérez's personal website, Software Developer based in Valencia (Spain)",
        "image": "https://fjp.es/logo.jpg",
        "twitterUsername": "@fjpalacios",
        "url": "https://fjp.es",
      },
      "resume": Object {
        "education": "Education",
        "experience": "Experience",
        "freelance": "Freelance work",
        "me": "About me",
        "projects": "Side projects",
        "talks": "Talks / workshops",
        "volunteering": "Volunteering",
      },
      "subtitle": "Software Developer",
      "title": "Francisco Javier Palacios Pérez",
    }
  }
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <TextArea
    text="Foo"
  >
    <div
      className="text-area"
    >
      <div
        className="text-area__content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Foo",
          }
        }
      />
    </div>
  </TextArea>
</IntlProvider>
`;
